### PR TITLE
Symlink native composer home

### PR DESCRIPTION
**Pull request by Slamdunk**
Reference: https://github.com/Chialab/docker-php/pull/31#issuecomment-367961738

### CODE REVIEW

**Comment by Nemo64** [comment]
looks good.

A few notes on the PATH variable:

* I think the PATH must be`$(composer config --global home)/vendor/bin:$PATH`. In the current implementation it would break if someone would mount the correct composer home since /root/composer would than be empty.
* A bit of a site note but shouldn't the path be appended to the path variable? If it is prepended, some rouge package could install binaries with the name `su` or `ssh` and silently catch your login information.

**Comment by Slamdunk** [comment]
> I think the PATH must be `$(composer config --global home)/vendor/bin:$PATH`. In the current implementation it would break if someone would mount the correct composer home since /root/composer would than be empty.

`$(composer config --global home)` notation doesn't work in `ENV` docker statement, see https://github.com/Chialab/docker-php/pull/31#issuecomment-367961738

> A bit of a site note but shouldn't the path be appended to the path variable? If it is prepended, some rouge package could install binaries with the name su or ssh and silently catch your login information.

I agree. I just changed according to previous implementation, let's listen what @fquffio thinks.

**Comment by Slamdunk** [comment]
To be honest I don't know if `$(...)` notation works in `RUN` statement either, I don't have the environment to test it at the moment.

**Comment by fquffio** [comment]
I didn't forget about this PR, but I hadn't time to take a look into it yet. I promise I'll do as soon as I can.

@Slamdunk `$(cmd)` does work in `RUN` instructions.

**Comment by Slamdunk** [comment]
> I didn't forget about this PR, but I hadn't time to take a look into it yet. I promise I'll do as soon as I can.

Free and open-source software maintainers aren't requested to respect a schedule.

Take all the time you need :wink:

File: 5.4/Dockerfile [hunk 1]
```diff
@@ -33,6 +33,6 @@ RUN buildDeps=" \
     && rm -r /var/lib/apt/lists/*
 
 # Install Composer.
-ENV COMPOSER_HOME /root/composer
 RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
-ENV PATH $COMPOSER_HOME/vendor/bin:$PATH
+RUN ln -s $(composer config --global home) /root/composer
```
Comment (fquffio) [inline 5.4/Dockerfile]: I'd try to chain this command to the previous `RUN`:
```Dockerfile
RUN curl ... && ln -s ...
```

File: 5.4/Dockerfile [hunk 2]
```diff
@@ -33,6 +33,6 @@ RUN buildDeps=" \
...
+ENV PATH /root/composer/vendor/bin:$PATH
```
Comment (fquffio) [inline 5.4/Dockerfile]: I agree with @Nemo64 that it would be better to have `$PATH:/root/composer/vendor/bin`, even if it used to be this way before.

**Comment by Slamdunk** [comment]
Tomorrow morning I'll do the changes

**Comment by Slamdunk** [comment]
Done

**Comment by fquffio** [comment]
Thank you @Slamdunk ! Sorry if it took such a long time for me to get this merged.

**Pull request merged**